version: "3.9"
name: pysect-backend-challenge-prod
services:

  db-prod:
    image: "postgres:13.10-alpine"
    container_name: pysect-backend-db-prod
    restart: unless-stopped
    ports:
      - "6666:5432"
    volumes:
      - pgdata-prod:/var/lib/postgresql/data
    networks:
      - nginx-reverse-proxy-net
    env_file:
      - ./docker/config/.env.prod

  redis-prod:
    image: redis:latest
    container_name: pysect-backend-redis-prod
    networks:
      - nginx-reverse-proxy-net

  web-prod:
    image: "simeonevilardo/pysect-backend-challenge-prod:latest"
    container_name: pysect-backend-challenge-prod
    platform: linux/arm64/v8
    volumes:
      - django-static-prod:/var/www/django/static
      - /home/pi/pysect-logs-prod:/var/www/log
    depends_on:
      - db-prod
    networks:
      - nginx-reverse-proxy-net
    env_file:
      - ./docker/config/.env.prod
    environment:
      VIRTUAL_HOST: py.sect.letz.dev
      VIRTUAL_PATH: ~^/(api|static)/
      VIRTUAL_PORT: 8000
    command: ./docker/django/local_run.sh

networks:
  nginx-reverse-proxy-net:
    external: true

volumes:
  pgdata-prod:
  django-static-prod:
